language/move-model/src/builder/module_builder.rs

```diff
@@ -1182,7 +1182,16 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
                                 "[internal] error in translating hlir type to prover type",
                             );
                         }
-                        et.define_local(loc, sym, ty, Some(Operation::Local(sym)));
+                        // TODO(refactoring): there appears to be a bug in how the Move compiler
```

Comment:
@tnowacki Here is where we currently work around the temporary bug. This should be removed and info.index directly used once this is fixed.

Reply:
Fix works and woraround removed.